### ISSUE TITLE
[FEATURE] Reindex user profile

### ISSUE THREAD

**Issue by gabbsmo** [pnp/powershell#3602]
**Describe the solution you'd like**
A cmdlet for triggering reindexing of user profiles

**Describe alternatives you've considered**
There is of course [this excellent sample](https://pnp.github.io/script-samples/spo-request-pnp-reindex-user-profile/README.html?tabs=pnpps), but it is a common task when building intranets so having it as a cmdlet in PnP-PowerShell should be considered.

**Comment by AndersRask** [comment]
Triggering "reindex" as it is called is really just touching the user profile by setting and unsetting a specific property.
I have never seen the use for this, as this will happen automatically for updated items anyway. So basically it puts unneeded pressure on the crawler as nothing has changed.
Also the script you mention performs quite poorly with items that has many user profile properties (+30k)

**Comment by gautamdsheth** [comment]
hey @gabbsmo , we have added a new dedicated cmdlet to upload photos in the latest nightly builds.
You can now use `Set-PnPUserProfilePhoto` to upload the photo and this photo is works across different M365 services quite fast without needing to reindex. It uses Graph API under the hood .

Can you please try it out and let us know ?

**Comment by reshmee011** [comment]
@gabbsmo : I’m closing this issue due to no response for over 1.5 years. Please feel free to open a new issue if you’d like to revisit it.

**Comment by gabbsmo** [comment]
@reshmee011 Sorry about the late reply. I don't see how `Set-PnPUserProfilePhoto` addresses the need to trigger search to reindex all user profiles. @gautamdsheth can you clarify?

**Comment by gautamdsheth** [comment]
hey @gabbsmo , PnP PowerShell already supports this. You can use `New-PnPUPABulkImportJob` or `Set-PnPUserProfileProperty`. Using either of these cmdlets would update the user profile property which would be picked by SharePoint indexer/crawler. 

New-PnPUPABulkImportJob can used for bulk updates while Set-PnPUserProfileProperty can be used for single user update, it is upto the user to decide how they want to use these cmdlets. Adding an additional cmdlet for this doesn't make sense IMHO.

**Comment by gabbsmo** [comment]
Yes, this is what the sample is doing. Still, there are a few steps involved. Getting all the users, build the json, observe the import job and some details inbetween.

There is already a convenience wrapper of the bulk import cmdlet that automatically builds json from given Entra Id extension properties. 

Thought there might be a demand for another for my use case. Obviously there is not a lot buzz in this issue so if you say it is not worth it - you are probably right and I appreciate your time and consideration.

**Comment by gautamdsheth** [comment]
For Set-PnPUserProfileProperty, you can simple use it in a loop.
For , the importjob cmdlet, there can different ways to fetch users and properties, creating JSON, such as , Entra ID , External systems like Workday , SharePoint Search etc., is too complex for this purpose.
Will close it for now , thanks for raising the request !
